libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/MapboxSpeechPlayer.java

```diff
@@ -243,12 +253,12 @@ private void startNextInstruction() {
 
   private void clearInstructionUrls() {
     while (!instructionQueue.isEmpty()) {
-      instructionQueue.remove().delete();
+      instructionQueue.poll();
```

Comment:
Why this change? you still need to `delete` the actual files, and you shouldn't have to worry about exceptions because of the `while(!instructionQueue.isEmpty())`

Reply:
The `poll()` Javadoc states `Retrieves and removes the head of this queue, or returns {@code null} if this queue is empty.` Does that not actually delete the file?  I changed it back in case I was missing context here.

Reply:
nope, it removes the File object from the queue and we lose track of it in memory but remember the audio file is still in the file system.